src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -399,12 +399,12 @@ const setLookAhead = async (args: SetLookAheadArgs): Promise<void> => {
       changeIndex: branch
     }
 
-    const getLastUsed = async (): Promise<number> =>
+    const getLastUsed = async (): Promise<number | undefined> =>
```

Comment:
Include the `?? -1` within `getLastUsed` and keep the return type as `Promise<number>`.

Reply:
Fixed.